### DISCUSSION TITLE
How to force turn in place?

### DISCUSSION THREAD

**Discussion by emaovo** [Sixze/ALS-Refactored#115] [Q&A]
Hi, 

I'm using the character in velocity rotation mode, so it rotates towards the direction of the velocity. 
there are moments in the game where I need to force the character to turn in place while still. 
What's the most elegant way to do this? Should I add a small movement input in the opposite direction, or is there a function I can simply call ? 

Thanks

**Comment by Sixze** [comment]
You can change the `UAlsAnimationInstance::IsTurnInPlaceAllowed()` function to something like this:

```cpp
bool UAlsAnimationInstance::IsTurnInPlaceAllowed()
{
	return RotationMode != AlsRotationModeTags::Aiming && ViewMode != AlsViewModeTags::FirstPerson;
}
```

**Reply by emaovo** [reply]
Thank you, I'll give a try. But this sorts of 'unlocks' the possibility to turn in place, but does not tell me how should I trigger it. 
Should I just call a "SetActorRotation" with the new rotation ?

**Reply by Sixze** [reply]
There is no explicit function to force a turn in place, but you can take a look at the `UAlsAnimationInstance::RefreshTurnInPlace()` and `UAlsAnimationInstance::PlayQueuedTurnInPlaceAnimation()` functions and create your own.
